sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,215 @@ class RotatingCertificateCredential implements TokenCredential {
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+
+const credential = new InteractiveBrowserCredential({
+  tokenCachePersistenceOptions: {
+    enabled: true
+  }
+});
+```
+
+#### Persisting the Authentication Record
+
+The `AuthenticationRecord` which is returned from the `authenticate`, contains data identifying an authenticated account. It is needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions. There is no sensitive data in the `AuthenticationRecord` so it can be persisted in a non-protected state. Ensure that you pass in the appropriate scopes for your service to `authenticate` method.
+
+Here is an example of an application storing the `AuthenticationRecord` to the local file system after authenticating the user.
+
+```ts
+import {
+  useIdentityExtension,
+  InteractiveBrowserCredential,
+  AuthenticationRecord,
+  serializeAuthenticationRecord
+} from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+import fs from "fs";
+import { promisify } from "util";
+import path from "path";
+
+useIdentityExtension(cachePersistenceExtension);
+
+export async function main(): Promise<void> {
+  const AUTH_RECORD_PATH = "./tokencache.bin";
+  const credential = new InteractiveBrowserCredential({
+    tokenCachePersistenceOptions: {
+      enabled: true
+    }
+  });
+  const authRecord: AuthenticationRecord = await credential.authenticate(
+    "https://service/.default"
+  );
+  const writeFileAsync = promisify(fs.writeFile);
+  const content = serializeAuthenticationRecord(authRecord);
+  await writeFileAsync(path.join(process.cwd(), AUTH_RECORD_PATH), content);
+}
+
+main().catch((err) => {
+  console.log("error code: ", err.code);
+  console.log("error message: ", err.message);
+  console.log("error stack: ", err.stack);
+});
+```
+
+#### Silent authentication with Authentication Record and Token Cache Persistence Options
+
+Once an application has configured a credential to persist token data and an `AuthenticationRecord`, it is possible to silently authenticate. This example demonstrates an application setting the `tokenCachePersistenceOptions` and retrieving an `AuthenticationRecord` from the local file system to create an `InteractiveBrowserCredential` capable of silent authentication.
+
+```ts
+import {
+  useIdentityExtension,
+  InteractiveBrowserCredential,
+  AuthenticationRecord,
+  deserializeAuthenticationRecord
+} from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+import fs from "fs";
+import { promisify } from "util";
+import path from "path";
+
+useIdentityExtension(cachePersistenceExtension);
+
+export async function main(): Promise<void> {
+  const AUTH_RECORD_PATH = "./tokencache.bin";
+  const readFileAsync = promisify(fs.readFile);
```

Comment:
Same comment about not overloading concepts with `promisify`.

Reply:
updated